### PR TITLE
feat(instrumentation-aws-sdk)!: Capture full ARN for span attribute messaging.destination.name for SNS topics

### PR DESCRIPTION
**Pull request by alingamn**
## Problem
The current approach of capturing topic names instead of ARNs in the messaging.destination attribute leads to a loss of valuable information regarding the account associated with the topic. This information is crucial for efficiently locating the topic.

## Solution
This PR includes the SNS topic arn as the span attribute 'messaging.destination.name' to uniquely identify the messaging destination.  Currently, 'messaging.destination.name' is added as a constant since it is not available as a semantic attribute yet.  Attribute messaging.destination.name is added as per  Semantic-convention https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/messaging.md

## Tests
Unit tests

Closes [#1716](https://github.com/open-telemetry/opentelemetry-js-contrib/issues/1716)

cc: @pavankrish123 @svrnm @noltak3 @yumarg

### CODE REVIEW

**Comment by pavankrish123** [comment]
cc: @dyladan please let us know if any further chAnges are needed

File: plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts [hunk 1]
```diff
@@ -41,7 +41,7 @@ export class SnsServiceExtension implements ServiceExtension {
         MessagingDestinationKindValues.TOPIC;
       const { TopicArn, TargetArn, PhoneNumber } = request.commandInput;
       spanAttributes[SemanticAttributes.MESSAGING_DESTINATION] =
```
Comment (pavankrish123) [inline plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts:43]: @alingamn  should the key be messaging.destination.name per latest spec?

Comment (alingamn) [reply plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts:43]: @pavankrish123 Semantic attribute messaging.destination.name is not yet available.  This is the PR to address that https://github.com/open-telemetry/opentelemetry-js-contrib/pull/1727

Comment (pavankrish123) [reply plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts:43]: Thanks Anusha

**Review by pavankrish123 — APPROVED** [review]

File: plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts (hunk 1 above)
Comment (pichlermarc) [reply plugins/node/opentelemetry-instrumentation-aws-sdk/src/services/sns.ts:43]: @alingamn We're having some trouble updating the semantic conventions package at the moment (they are auto-generated but already released as stable). In the meantime, please add any non-stable semantic attribute names as constants here in this package.

**Review by pichlermarc — APPROVED** [review]
Overall LGTM, but I would really like either @blumamir or @carolabadeer to also take a look as they're the component owners.

**Comment by alingamn** [comment]
> Overall LGTM, but I would really like either @blumamir or @carolabadeer to also take a look as they're the component owners.

@blumamir or @carolabadeer It would be great if one of you could take a look at this PR

**Comment by pavankrish123** [comment]
@wangzlei would you be able to review it from AWS

**Comment by pavankrish123** [comment]
cc: @srprash can please get a review on JS side as well.

**Comment by pavankrish123** [comment]
@pichlermarc  would you be able to approve the PR now as Python side changes has been +1 ed from AWS X-Ray team? Please and thank you.

**Comment by dyladan** [comment]
Using arn is previously agreed

**Comment by dyladan** [comment]
Running tests and I will merge

**Comment by dyladan** [comment]
This is a breaking change. Please update the pr title with the appropriate semantic commits ! to reflect that

**Comment by alingamn** [comment]
> This is a breaking change. Please update the pr title with the appropriate semantic commits ! to reflect that

@dyladan  I have updated the code to add a new span attribute 'messaging.destination.name as constant since it's not available as in the semantic-conventions repo yet.  span attribute messaging.destination.name should be used as per the semantic- convention https://github.com/open-telemetry/semantic-conventions/blob/main/docs/attributes-registry/messaging.md.  Could you please run the tests again and Merge the PR? Thank you

**Comment by dyladan** [comment]
This is a breaking change. Please update the pr title with the appropriate conventional commit message. Adding a ! Should be sufficient

**Comment by pavankrish123** [comment]
@dyladan any chance we can get this merged?

**Pull request merged**